backend/src/main/java/com/greenfoxacademy/backend/controller/UserController.java

```diff
@@ -76,4 +77,25 @@ public ResponseEntity<ProfileUpdateResponseDto> userProfileUpdate(
       return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
     }
   }
+
+  /**
+   * This method delete a user from the database.
+   *
+   * @param principal the user that is authenticated
+   * @return a response entity with the status code
+   */
+  @DeleteMapping("/delete-profile")
+  public ResponseEntity<?> deleteProfile(Principal principal) {
+    try {
+      if (principal == null) {
+        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body("User is not authenticated");
+      }
```

Comment:
This thing should be redundant. You will never know unless you test it manually or even automated ways.

Reply:
Resolved on https://github.com/gfa-cc-after/Pet-Clinic-GilmoreDevs/pull/75/commits/8401f3b2c5d008f8f14ce5f3480b1f450849b331